how-to-release.md

```diff
@@ -0,0 +1,274 @@
+
+When a new release branch needs to be created, you first need to tag the forking point on master branch and then create the release branch named as `rX.Y`. Assuming for the sake of example that the new release is `0.3.0`, the detailed commands needs to be executed.
+
+```
+# Tag the branch cut position with `branch-X.Y`
+git checkout master
+git tag branch-0.3
+git push origin branch-0.3
+
+# Create the release branch `rX.Y`
+git checkout -b r0.3
+git push origin r0.3
+```
+
+Now the release branch is ready to go.
+
+# Build a release candidate
+
+When the community has the consensus to start a release candidate without any concerns or objections on open release-blocking issues, you can start to build a release candidate.
+
+Pravega release managers should keep in mind that the core of the release process is the build-vote-fix cycle. Each cycle produces one release candidate. The Release Manager repeats this cycle until the community approves one release candidate, which is then finalized.
+
+To start a release candidate, you now need to do several things:
+- Create and tag the release candidate on github
+- Deploy jar artifacts to the Sonatype Repository, which is the staging area for deploying the jars to Maven Central
+
+## Tagging the candidate on Github
+First make sure you work on the release branch named `rX.Y`, e.g, `r0.3`. 
+
+Before you execute any command, double check and keep these two things in mind.
+- What version we want to release as `X.Y.Z`, e.g, `0.3.0`
+- The release candidate number `N`. The number `N` starts with 0 and increment for each candidate.
+
+Next, follow the below steps carefully to start a release candidate, here we still take `0.3.0` release as an example, so you need to replace with the correct version and number in some places.
+
+1. Create a new _temporary_ release branch on the latest commit of `rX.Y` branch, named as `release-X.Y.Z-rcN`, e.g. `release-0.3.0-rc0`. (The "temporary" here means that you need to clean up the branch after the process finishes)
+
+```
+// Make sure you are on the right commit
+git checkout -b release-0.3.0-rc0
+```
+
+2. Remove the `-SNAPSHOT` suffix in all the versions of Pravega projects in `gradle.properties`. For example, if the current value is `pravegaVersion=0.3.0-SNAPSHOT`, then change it to `pravegaVersion=0.3.0`.
+
+3. Push this commit onto the release branch. The commit message should be `Add release X.Y.Z`. A pull request is NOT required for this commit.
+```
+git commit -s -am "Add release 0.3.0"
+git push origin release-0.3.0-rc0
+```
+
+4. Tag the commit with `vX.Y.Z-rcN` and push this tag. For example, `v0.3.0-rc0`.
+```
+git tag v0.3.0-rc0
+git push origin v0.3.0-rc0
+``` 
+
+5. Delete the temporary release branch.
+```
+git branch -D release-0.3.0-rc0
+```
+
+## Staging the binary releases
+* Build with the following command, there could be some slight differences regarding the parameters and plugin used for publishing.
```

Comment:
we should stage the docker images (see my comment below)

Reply:
Added the steps to stage the images